pkg/model/mask.go

```diff
@@ -115,6 +110,50 @@ func updateField(src, dst reflect.Value, fields []string) {
 	}
 }
 
+// updateSliceField this function updates a destination slice by either replacing a struct with a matching ID
+// or appending new values if no matching ID is found.
+func updateSliceField(srcField, dstField reflect.Value) {
+	appendOnly := func() bool {
+		el := srcField.Type().Elem()
+
+		if el.Kind() != reflect.Struct {
+			return true
+		}
+
+		_, ok := el.FieldByName("ID")
+
+		return !ok
+	}
+
+	if appendOnly() {
+		// If the field is a slice, append new values from src to existing slice in dst
+		//nolint:lll
+		// https://github.com/protocolbuffers/protobuf/blob/9bbea4aa65bdaf5fc6c2583e045c07ff37ffb0e7/src/google/protobuf/field_mask.proto#L111
+		dstField.Set(reflect.AppendSlice(dstField, srcField))
```

Comment:
We can remove this comment but move the link to protobuf description to func description.
```suggestion
		dstField.Set(reflect.AppendSlice(dstField, srcField))
```